recipes/gcc/all/conanfile.py

```diff
@@ -155,11 +158,183 @@ def package(self):
-        if self.settings.os in ["Linux", "FreeBSD"]:
-            self.cpp_info.system_libs.append("m")
-            self.cpp_info.system_libs.append("rt")
-            self.cpp_info.system_libs.append("pthread")
-            self.cpp_info.system_libs.append("dl")
+        # FIXME: Add more detailed triplet identification as per https://github.com/conan-io/conan/issues/12789
+        # os_build, arch_build, os_host, arch_host = get_cross_building_settings(self)
+        # compiler = self.settings.get_safe("compiler")
+        # # e.g., x86_64-pc-linux-gnu
+        # triplet = _get_gnu_triplet(os_host, arch_host, compiler=compiler)
+        triplet = "x86_64-pc-linux-gnu"
+
+        # Libs
+        # Shared            | Static
+        # ==================|====================
+        # libitm.so         | libitm.a              # GNU Transactional Memory Library. Transaction support for accesses to a process' memory, enabling easy-to-use synchronization of accesses to shared memory by several threads
+        # libgfortran.so    | libgfortran.a         # GNU Fortran Library
+        # libgomp.so        | libgomp.a             # GNU Offloading and Multi-Processing Project
+        # libubsan.so       | libubsan.a            # Undefined Behaviour sanitizer
+        # libtsan.so        | libtsan.a             # Thread sanitizer
+        # libstdc++.so      | libstdc++.a           # C++ Standard library
+        # libgcc_s.so       |                       # Dynamic gcc runtime. Combination of libgcc and libgcc_eh? Used by -shared-libgcc
+        # libcc1.so         |                       # GCC cc1 plugin for gdb
+        # liblsan.so        | liblsan.a             # Leak sanitizer
+        # libasan.so        | libasan.so            # Address sanitizer
+        # libssp.so         | libssp.a              # Stack Smashing Protector library
+        # libquadmath.so    | libquadmath.a         # GCC Quad Precision Math Library
+        # libcp1plugin.so   |                       # Library interface to C++ frontend
+        # libcc1plugin.so   |                       # Library interface to C frontend
+        # libatomic.so      | libatomic.a           # GNU atomic library
+        # liblto_plugin.so  |                       # Link time optimization plugin
+        #                   | libsupc++.a           # A subset of libstdc++.a. Contains only support routines defined by clause 18 of the standard.
+        #                   | libstdc++fs.a         # Experimental extension for std::filesystem
+        #                   | libssp_nonshared.a    #
+        #                   | libgcc.a              # Static gcc runtime. Used by -static-libgcc
+        #                   | libcaf_single.a
+        #                   | libgcc_eh.a           # libgcc exception handling. User by -static-libgcc
+        #                   | libgcov.a             # test coverage library
+
+        self.cpp_info.set_property("cmake_target_name", "gcc::gcc_all")
```

Comment:
Most of the targets are the dame ad the default and I am pretty sure they aren't from upstream so you can save some complexity and remove them

Reply:
Can you elaborate what you mean? I've modelled the libraries as components mostly to represent the relationships between them so that they can be included in the link command when necessary and linked in the right order - my understanding is that `collect_libs` wouldn't satisfy this.

This specific line I felt necessary to put in because the default catchall target `gcc::gcc` was overridden to refer to `libgcc.a`